beacon-chain/casper/validator.go

```diff
@@ -143,27 +144,29 @@ func ValidatorShardID(pubKey []byte, validators []*pb.ValidatorRecord, shardComm
 	return 0, fmt.Errorf("can't find shard ID for validator with public key %#x", pubKey)
 }
 
-// ValidatorSlotAndResponsibility returns a validator's assingned slot number
+// ValidatorSlotAndRole returns a validator's assingned slot number
 // and whether it should act as an attester or proposer.
-func ValidatorSlotAndResponsibility(pubKey []byte, validators []*pb.ValidatorRecord, shardCommittees []*pb.ShardAndCommitteeArray) (uint64, string, error) {
+func ValidatorSlotAndRole(pubKey []byte, validators []*pb.ValidatorRecord, shardCommittees []*pb.ShardAndCommitteeArray) (uint64, pbrpc.ValidatorRole, error) {
```

Comment:
All I did here was return the actual enum instead of a string, since callers of this function were always converting the enum to the string anyways.